### PR TITLE
Add some more tests

### PR DESCRIPTION
**Pull request by M4RKUS-11111**

### CODE REVIEW

**Comment by 6543** [comment]
It would be nice if you can add some more versions with RC and git infromation to **version_collection_test.go** too :+1:

File: constraint_test.go [hunk 1]
```diff
@@ -15,6 +15,8 @@ func TestNewConstraint(t *testing.T) {
 		{">= 1.x", 1, false},
 		{">= 1.2, < 1.0", 2, false},
 
+		//{"-1.0", 1, true},?
+
```
Comment (M4RKUS-11111) [inline constraint_test.go]:
```suggestion
```

Comment (M4RKUS-11111) [inline constraint_test.go:19]:
```suggestion
```

**Comment by 6543** [comment]
thx

**Review by 6543 — APPROVED** [review]

**Pull request merged**